pkg/codegen/internal/test/testdata/simple-enum-schema/dotnet/Enums.cs

```diff
@@ -0,0 +1,47 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+using System.Runtime.Serialization;
+
+namespace Pulumi.PlantProvider
+{
+    /// <summary>
+    /// plant container colors
+    /// </summary>
+    public readonly struct ContainerColor : IEquatable<ContainerColor>
```

Comment:
Have you had a chance to think much about how the serialization/deserialization is going to work?

We likely need to add a marker attribute [in the SDK](https://github.com/pulumi/pulumi/blob/master/sdk/dotnet/Pulumi/Serialization/Attributes.cs), something like:

```c#
namespace Pulumi
{
    [AttributeUsage(AttributeTargets.Class)]
    public sealed class EnumTypeAttribute : Attribute
    {
    }
}
```

That we can then annotate on these like:

```c#
[EnumType]
public readonly struct ContainerColor : IEquatable<ContainerColor>
{
    ...
}
```

So we can detect these during serialization/deserialization and do the proper thing to convert to/from the underlying string.

We'd only have to annotate these structs... the ones that are actual enums (e.g. `public enum ContainerSize`), we can already detect via [`Type.IsEnum`](https://docs.microsoft.com/en-us/dotnet/api/system.type.isenum?view=netcore-3.1).

Let me know if you need any help here, I can throw together a rough draft of what the serialization/deserialization looks like for .NET.

Reply:
Yeah, I started working on that yesterday. And yeah detecting the actual enums is pretty straightforward, thanks for the tip on the structs! I'm gonna work through the SDK work first before coming back to this.